presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDriverUri.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.jdbc;
+
+import com.google.common.base.Splitter;
+import com.google.common.net.HostAndPort;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static io.airlift.http.client.HttpUriBuilder.uriBuilder;
+
+/**
+ * Container for parameters of a JDBC connection. The class is also responsible for parsing Presto JDBC URL.
+ */
+final class PrestoDriverUri
+{
+    private static final String JDBC_URL_START = "jdbc:";
+
+    private static final Splitter QUERY_SPLITTER = Splitter.on('&').omitEmptyStrings();
+    private static final Splitter ARG_SPLITTER = Splitter.on('=').limit(2);
+
+    private final HostAndPort address;
+    private final URI uri;
+
+    private String catalog;
+    private String schema;
+
+    public PrestoDriverUri(String url)
+            throws SQLException
+    {
+        this(parseDriverUrl(url));
+    }
+
+    public PrestoDriverUri(URI uri)
+            throws SQLException
+    {
+        this.uri = uri;
+        this.address = HostAndPort.fromParts(uri.getHost(), uri.getPort());
+
+        Map<String, String> params = parseParameters(uri.getQuery());
+
+        initCatalogAndSchema();
+    }
+
+    public URI getURI()
```

Comment:
Let's call this `jdbcUri` so it's clear this is the original JDBC URI (vs an HTTP URI for the Presto server)